src/agent/onefuzz-agent/src/local/common.rs

```diff
@@ -275,17 +273,13 @@ impl DirectoryMonitorQueue {
         let handle: tokio::task::JoinHandle<Result<()>> = tokio::spawn(async move {
             let mut monitor = DirectoryMonitor::new(directory_path_clone.clone());
             monitor.start()?;
-            loop {
-                match monitor.poll_file() {
-                    Poll::Ready(Some(file_path)) => {
-                        let file_url = Url::from_file_path(file_path)
-                            .map_err(|_| anyhow!("invalid file path"))?;
-                        queue.enqueue(file_url).await?;
-                    }
-                    Poll::Ready(None) => break,
-                    Poll::Pending => delay_with_jitter(Duration::from_secs(1)).await,
-                }
+
+            while let Some(file_path) = monitor.next_file().await {
```

Comment:
Double-check that this is an equivalent or improved substitution. I think this is what the old code really wanted. The only thing lost here is the jittered delay, but I think we'd prefer yielding to the async runtime vs. manually polling anyway.

Reply:
I agree with your assessment.

Reply:
looks good